Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIBookmarksCocoa.mm

```
@@ -251,22 +295,18 @@
 
 void WebExtensionAPIBookmarks::getTree(Ref<WebExtensionCallbackHandler>&& callback, NSString **outExceptionString)
 {
-    if (m_mockBookmarks.isEmpty())
-        initializeMockBookmarksInternal();
-
-    auto rootNodeOptional = m_mockBookmarks.getOptional(bookmarksRootId);
-    if (!rootNodeOptional) {
-        if (outExceptionString)
-            *outExceptionString = toErrorString(nullString(), nullString(), @"root not found").createNSString().autorelease();
-        return;
-    }
-
-    if (auto rootOptional = m_mockBookmarks.getOptional(bookmarksRootId)) {
-        Ref finalTreeRoot = rootOptional.value();
-        NSDictionary *rootDict = toWebAPI(finalTreeRoot);
-        NSArray* resultArray = [NSArray arrayWithObject:rootDict];
-        callback->call(resultArray);
-    }
+    WebProcess::singleton().sendWithAsyncReply(
+        Messages::WebExtensionContext::BookmarksGetTree(),
+        [protectedThis = Ref { *this }, callback = WTFMove(callback)](Expected<WebExtensionBookmarksParameters, WebExtensionError>&& result) {
+            if (!result)
+                return;
```

We should be calling the callback with the error here, rather than just swallowing it